### PR TITLE
Add 'connects-to' annotation

### PR DESCRIPTION
**Pull request by sbose78**
This annotation used to denote that a component A might be talking to Component B.

Screen Shot 2019-04-12 at 1 03 04 PM

### CODE REVIEW

**Comment by DhritiShikhar** [comment]
@sbose78 
[1] why is this `connects-to` an annotation and not a label?
[2] Can any component have more than one `connects-to` annotation?
[3] What is the string value that `connects-to` uses? Is it the `app.kubernetes.io/name` value or `app.kubernetes.io/instance` value?

**Comment by sbose78** [comment]
> What is the string value that connects-to uses? Is it the app.kubernetes.io/name value or app.kubernetes.io/instance value?

`instance` preferably. but not necessarily. 

> Can any component have more than one connects-to annotation?

Yes.

> why is this connects-to an annotation and not a label?

Because `connects-to` value is not to be used to actually create a connection.

**Comment by sspeiche** [comment]
LGTM

File: labels-annotation-for-openshift.adoc [hunk 1]
```diff
@@ -113,6 +113,11 @@ TIP: Labels must be applied to ALL resources, DeploymentConfigs, BuildConfigs, S
 | Identifier for the version of the source code. Can be branch, tag or commit SHA for the application. If omitted master head of the default branch is assumed
 | v1.0.0
 
+| app.openshift.io/connects-to:
+| OPT
+| The component that this component is connecting to (e.g. a component making API calls to another `service` ) 
+| e.g. Component `front-end` has a `connects-to` which has a value `backend`
+
```
Comment (gorkem) [inline labels-annotation-for-openshift.adoc:120]: I am not sure if we are clear about what value is expected on this definition. Is it `app.kubernetes.io/instance` or `app.kubernetes.io/name` or a combination of both?

File: labels-annotation-for-openshift.adoc [hunk 2]
```diff
@@ -113,6 +113,11 @@ TIP: Labels must be applied to ALL resources, DeploymentConfigs, BuildConfigs, S
 | Identifier for the version of the source code. Can be branch, tag or commit SHA for the application. If omitted master head of the default branch is assumed
 | v1.0.0
 
+| app.openshift.io/connects-to:
+| OPT
+| The component that this component is connecting to (e.g. a component making API calls to another `service` ) 
+| e.g. Component `front-end` has a `connects-to` which has a value `backend`
```
Comment (gorkem) [inline labels-annotation-for-openshift.adoc:119]: I am not sure about this example. Can we try to put some real values?

**Review by gorkem — CHANGES_REQUESTED** [review]

**Review by gorkem — APPROVED** [review]

**Pull request merged**